src/ebay-carousel/carousel-item.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {
+    ComponentProps,
+    FC,
+    RefObject,
+    useImperativeHandle,
+    useRef,
+    useState
+} from 'react'
+import { withForwardRef } from '../common/component-utils'
+import { getRelativeRects } from './helpers'
+import { ListItemRef } from './types'
+
+type ListProps = ComponentProps<'li'>
+type CarouselItemProps = ListProps & {
+    slideWidth?: number;
+    offset?: number;
+    className?: string;
+    forwardedRef?: RefObject<ListItemRef>
+};
+
+
+const CarouselItem: FC<CarouselItemProps> = ({ slideWidth, offset, forwardedRef, children, ...rest }) => {
+    const itemRef = useRef()
+    const [isVisible, setIsVisible] = useState(false)
+
+    useImperativeHandle(forwardedRef, () => {
```

Comment:
If we remove the CarouselItem and CarouselList components and move everything inside the Carousel component, do we still need to use `useImperativeHandle`? I feel that we are creating here an explicit dependency between this hook and how we use this information and also we add this properly on the ref object it self making the logic complex to understand. I feel that if we move the rendering `ul` and `li` inside the Carousel it self might be easier since we have the reference of the items in there. (Might be wrong, just trying to simplify the logic here)

Reply:
If we remove the CarouselItem and CarouselList, there won't be a need of using `useImperativeHandle` but then we extend the logic in one component which will make it very hard to read, and second how we will render the carousel items? We need to have a component for adding carousel items! @darkwebdev what do you think?

Reply:
Sometimes early abstraction adds more complexity than simplify them. We are creating an implicit dependency between this component and how this is being used. Also number of line doesn't define complexity, relations, new interfaces and the need to jump around in the code adds complexity. I think that we can leave like this for now and review that later.